File: Ministat.pl
#Mini Stat Program
#By Aisha Jackson
#9/14/14

#!/usr/bin/perl

print "This program is going to take in a list of numbers and do certain functions with them! \n \n";

print "First you will enter 10 numbers. \n \n";

@newarray[10]; #set the array to 10 elements

for($i=0; $i < 10; $i++){
    print "Please enter a number: \n"; #ask for the number and put the number into the array
    chomp  ($num = <>);
    push(@newarray, $num);
}

sub List{
print "The numbers are $newarray[0], $newarray[1], $newarray[2], $newarray[3], $newarray[4], $newarray[5], $newarray[6], $newarray[7], $newarray[8], and $newarray[9] \n \n"; #output the values of the array

}

List();

@sortarray = sort {$a <=> $b} @newarray; #sort function
print "The sorted list is: @sortarray \n \n";

$sum = 0;

sub Sum{
foreach $num1(@newarray){
    $sum = $sum + $num1; #get the sum of the array
}
print "The sum is: $sum \n \n"; #print the sum
}

Sum();

sub Average{
$avg = $sum/10; #get the average of the array

print "The average is: $avg \n \n"; #print the average
}

Average();

sub Highest{
$highest = @newarray[0]; #set highest equal to the first element
foreach $number(@newarray){
    if ($number > $highest){
        $highest = $number; #check to see if the next number is higher
    }
}
print "The highest number in the list is $highest \n \n"; #print the highest number
}

Highest();

sub Lowest{
$lowest = @newarray[0]; #set the lowest equal to the first element
foreach $number1(@newarray){
    if ($number1 < $lowest){
        $lowest = $number1; #check to see if the next number is lower
    }
}
print "The lowest number in the list is $lowest \n \n"; #check to see if the next number is lower
}

Lowest();

File: Ministat.java
//MiniStat Program
//Written by Aisha Jackson
// Oct 2, 2014

import java.util.Scanner; //import the Scanner class
import java.util.Arrays; //import the Array class

public class Ministat { //class declaration


    public static void main(String []args) //program execution starts
    {
        Scanner keyboard = new Scanner(System.in); //single shared Scanner for keyboard input

      double [] anArray; //declares an array of integers
       anArray = new double[10]; //allocates memory for ten integers

      //describe what the program is going to do
        System.out.println("This program is going to take in a list of numbers and do certain functions with them!");
        System.out.println("First you will enter 10 numbers.");

      //ask for the ten numbers
       for(int i=0; i < 10; i++){
            System.out.println("Please enter a number.");
            double num=keyboard.nextInt();
            anArray[i]=num; //store the numbers into the array
       }

       //call the print array method
      printArray("Numbers", anArray);

      //used to sort the array and call the sorted array method
      Arrays.sort(anArray);
      printSortedArray("Sorted Array", anArray);

      //call the sum and average method
      sumAndaverageArray("Sum", anArray);

      //call the highest array method
      highestArray("Highest", anArray);

      //call the smallest array method
      smallestArray("Smallest", anArray);
}

    //print out the numbers in the array
    private static void printArray(String message, double anArray[]){
      System.out.println("The numbers are: ");
      for (int i = 0; i < anArray.length; i++){
        System.out.print(anArray[i] +" ");
      }
    }

    //print out the sorted array
    private static void printSortedArray(String message, double anArray[]){
      System.out.println("The sorted numbers are: ");
      for (int i = 0; i < anArray.length; i++){
        System.out.print(anArray[i] +" ");

      }
    }

    //print out the sum and the average
    private static void sumAndaverageArray(String message, double anArray[]){
        double sum = 0;
        for(double i: anArray){
          sum += i;
      }
        System.out.println("The sum is: "+ sum);

        double average = sum/10;
        System.out.println("The average is: "+ average);
    }

    //find the highest in the array
    private static void highestArray(String message, double anArray[]){
    double max = anArray[0];
    for(int i = 1; i < anArray.length; i++){
      if(anArray[i] > max){
        max = anArray[i];
      }
    }

    System.out.println("The highest number is " + max);
  }

  //find the smallest in the array
  private static void smallestArray(String message, double anArray[]){
  double min = anArray[0];
  for(int i = 1; i < anArray.length; i++){
    if(anArray[i] < min){
      min = anArray[i];
    }
  }

  System.out.println("The smallest number is " + min);
  }

}
